src/Config/RuntimeConfigLoader.cs

```diff
@@ -101,4 +116,57 @@ public static JsonSerializerOptions GetSerializationOptions()
         options.Converters.Add(new StringJsonConverterFactory());
         return options;
     }
+
+    /// <summary>
+    /// It adds or replaces a property in the connection string with `Application Name` property.
+    /// If the connection string already contains the property, it uses a regular expression to update the existing value
+    /// by adding the DataApiBuilder Application Name (dab_oss/dab_hosted). If not, it appends the property `Application Name` to the connection string.
+    /// This method only adds the `Application Name` property to specify DataApiBuilder Application Name.
+    /// </summary>
+    /// <param name="connectionString">Connection string for connecting to database.</param>
+    /// <returns>Updated connection string with `Application Name` property.</returns>
+    internal static string GetConnectionStringWithApplicationName(string connectionString)
+    {
+        // If the connection string is null, empty, or whitespace, return it as is.
+        if (string.IsNullOrWhiteSpace(connectionString))
+        {
+            return connectionString;
+        }
+
+        // Get the application name using ProductInfo.GetDataApiBuilderUserAgent().
+        string applicationName = ProductInfo.GetDataApiBuilderUserAgent();
+
+        // Create a StringBuilder from the connection string.
+        StringBuilder stringBuilder = new(connectionString);
```

Comment:
Can we use ConnectionStringBuilder object? and set the ApplicationName property of that object? That way we dont need to do any string comparison checks ourselves...?

Reply:
Thanks, using ConnectionStringBuilder made the code simpler.